Dockerfile

```diff
@@ -42,24 +47,31 @@ RUN ollama serve & while ! curl http://localhost:11434; do sleep 1; done; ollama
 # COPY --chmod=644 Modelfile Modelfile
 # RUN curl --location https://huggingface.co/TheBloke/DiscoLM_German_7b_v1-GGUF/resolve/main/discolm_german_7b_v1.Q5_K_S.gguf?download=true --output discolm_german_7b_v1.Q5_K_S.gguf; ollama serve & while ! curl http://localhost:11434; do sleep 1; done; ollama create ${MODEL} -f Modelfile && rm -rf /tmp/model
 
+
 # Setup the custom API and frontend
 WORKDIR /workspace
-COPY --chmod=644 gswikichat gswikichat
-COPY --chmod=755 frontend frontend
-COPY --chmod=755 json_input json_input
 
-# Install node from upstream, ubuntu packages are too old
-RUN curl -sL https://deb.nodesource.com/setup_20.x | bash
-RUN apt-get install -y nodejs && \
-    apt-get clean && \
-    rm -rf /var/lib/apt/lists/*
+# Install backend dependencies
+COPY --chmod=755 requirements.txt requirements.txt
+RUN pip install -r requirements.txt
 
-# Install node package manager yarn 
-RUN npm install -g yarn
 
-# Install frontend dependencies and build it for production (into the frontend/dist folder)
-RUN cd frontend && yarn install && yarn build
+# Install frontend dependencies
+COPY --chmod=755 frontend/package.json frontend/package.json
+COPY --chmod=755 frontend/yarn.lock frontend/yarn.lock
+RUN cd frontend && yarn install
+
+
+# Copy and build frontend for production (into the frontend/dist folder)
+COPY --chmod=755 frontend frontend
+RUN cd frontend && yarn build
```

Comment:
I am curious why the Dockerfile changes directory to frontend twice

Reply:
Actually, I commented out both of these lines and kept the original towards the end after installing node18 and, thus, yarn.

```Dockerfile
# RUN cd frontend && yarn install
...
# RUN cd frontend && yarn build
...
RUN cd frontend && yarn install && yarn build
...
# Container startup script
COPY --chmod=755 start.sh /start.sh
CMD [ "/start.sh" ]
```

Reply:
> I am curious why the Dockerfile changes directory to frontend twice

Docker build is weird, each `RUN` is happening in a new container, so everything from the last `RUN` (processes, working directory) is lost. That's why every `RUN` changes the directory individually. What would work though is `WORKDIR`, this remains set throughout the `Dockerfile` including in the resulting container. I think it is a matter of taste what to use in the build process.

Reply:
> Actually, I commented out both of these lines and kept the original towards the end after installing node18 and, thus, yarn. (See below)

I split `yarn install` and `yarn build` intentionally into two `RUN`s in order to use the docker build cache effectively. `yarn install` is only repeated when dependencies in `package.json` / `yarn.lock` change, `yarn build` is repeated when frontend source files change.

Reply:
@exowanderer Could you please see if #22 also builds on your machine? This is basically this branch, but downgraded to node 18.